Adding a telemetry page (#544)
ArcadeAI/docs: app/en/references/mcp/telemetry/page.mdx

---
title: "Arcade MCP Telemtry"
description: "Learn about what data we track when using arcade-mcp"
---

# Telemetry

> Arcade collects telemetry data about general usage of `arcade-mcp` , our open source MCP development framework. Participation in this program is optional, and you may opt-out if you'd not like to share any information.

Reviewer: ```suggestion
Arcade collects telemetry data about the usage of `arcade-mcp`, our open source MCP development framework. Participation in this program is optional, and you may opt-out if you'd not like to share any information.
```

> Note: This page covers telemetry for local and self-hosted `arcade-mcp` only. Our cloud products collect different usage data for improvement and troubleshooting as part of the service, as defined in our [terms-of-service](https://www.arcade.dev/terms-of-service). Commands that connect you to your Arcade account (`arcade deploy`, `arcade login`) also require accepting our [terms-of-service](https://www.arcade.dev/terms-of-service), which supersede this document:

Reviewer: ```suggestion
Note: This page covers telemetry for local and self-hosted use of `arcade-mcp` only. Our cloud products collect different usage data for improvement and troubleshooting as part of the service, as defined in our [terms-of-service](https://www.arcade.dev/terms-of-service). Commands that connect you to your Arcade account (`arcade deploy`, `arcade login`) also require accepting our [terms-of-service](https://www.arcade.dev/terms-of-service), which supersedes this document:
```

> - `arcade login `associates tool calls with your identity for authentication, but doesn’t track which tools you use or their parameters

Reviewer: ```suggestion
- `arcade login` associates CLI commands and tool calls with your identity for authentication and secret management
```

- `arcade deploy `deploys to Arcade’s hosted infrastructure where full telemetry is collected per our [terms-of-service](https://www.arcade.dev/terms-of-service).

#### Why is telemetry collected?

Arcade has been growing quickly since its release. Without telemetry collection, we make decisions about how to improve our products through dogfooding of our MCP Servers, tools, and auth providers. We also actively engage with the community to gather feedback through Discord, YouTube, LinkedIn, and X.

> However, this approach only allows us to collect feedback from a subset of developers. This subset may have different needs and use cases than you. Telemetry allows us to accurately gauge feature usage, identify pain points, and understand how developers use our local MCP products.

Reviewer: ```suggestion
However, this approach only allows us to collect feedback from a subset of developers. This subset may have different needs, environments, and use cases than you. Telemetry allows us to accurately gauge feature usage, identify pain points, and understand how developers use our local MCP products.
```

This data will let us better tailor Arcade to the masses, ensuring its continued growth, relevance, and best-in-class developer experience. Furthermore, this allows us to **verify whether improvements made to our product are actually working for all users.**

#### What is being collected?

We track general usage information for the `arcade-mcp` product, including:

- Command invoked (e.g., arcade new, arcade evals, other commands listed [here](https://docs.arcade.dev/en/home/arcade-cli#arcade-new) )

> - Count of local tool executions (we do NOT collect tool names, parameters, or any other personally identifiable information related to the tools you or your agents execute locally)

Reviewer: ```suggestion
- Count of local tool executions (we do NOT collect tool names, parameters, or any other personally identifiable information related to the tools you or your agents execute locally or when self-hosted)
```

- General machine information (e.g., macOS/Windows/Linux, whether the command was run within the CLI)
- Duration of commands
- Errors thrown

Note: this list is regularly audited to ensure its accuracy

#### What about sensitive data (e.g. secrets)?

When running `arcade-mcp `outside of our cloud service, we do not collect any metrics which may contain sensitive data.

This includes, but is not limited to:

- Tool names or which tools you're using
- Tool parameters or arguments
- Environment variables
- Logs
- Any personally identifiable information (PII)

#### How do I opt-out?

You may opt-out by setting the ARCADE_USAGE_TRACKING environment variable.

For example, to opt out for the lifetime of your terminal session, you can enter

> ```

Reviewer: ```suggestion
```bash
```
